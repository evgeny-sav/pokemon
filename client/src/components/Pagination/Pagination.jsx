import React from 'react';
import { useDispatch } from 'react-redux';
import { Pagination } from 'react-bootstrap';
import { fetchNext, fetchPrev } from '../../store/actions/pokemonsActions';

// eslint-disable-next-line react/prop-types
const PaginationBar = ({ previousLink, nextLink, page, setPage, count }) => {
  const dispatch = useDispatch();
  return (
    <Pagination>
      <Pagination.Item
        disabled={!previousLink}
        onClick={() => {
          setPage(page - 1);
          return previousLink && dispatch(fetchPrev(previousLink));
        }}
      >
        Prev
      </Pagination.Item>
      <Pagination.Item>{`Page ${page} of ${Math.ceil(
        count / 20,
      )}`}</Pagination.Item>
      <Pagination.Item
        disabled={!nextLink}
        onClick={() => {
          setPage(page + 1);
          return nextLink && dispatch(fetchNext(nextLink));
        }}
      >
        Next
      </Pagination.Item>
    </Pagination>
  );
};

export default PaginationBar;
