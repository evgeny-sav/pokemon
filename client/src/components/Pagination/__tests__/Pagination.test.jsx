import React from 'react';
import { create } from 'react-test-renderer';
import Pagination from '../Pagination';

describe('Pagination', () => {
  it('should render', () => {
    const component = create(<Pagination />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
