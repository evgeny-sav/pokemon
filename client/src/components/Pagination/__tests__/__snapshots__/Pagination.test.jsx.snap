// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item disabled"
  >
    <span
      className="page-link"
      disabled={true}
      onClick={[Function]}
    >
      Prev
    </span>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      Page undefined of NaN
    </a>
  </li>
  <li
    className="page-item disabled"
  >
    <span
      className="page-link"
      disabled={true}
      onClick={[Function]}
    >
      Next
    </span>
  </li>
</ul>
`;
